File: server/src/Models/Company.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;
using System.Reflection.Metadata;

namespace server.src.Models
{
    public class Company
    {
        [Key]
        public string BusinessId { get; set; } = null!;
        public string Name { get; set; } = null!;
        public DateOnly RegistrationDate { get; set; }
        public CompanyFormType? CompanyForm { get; set; }
        public string? DetailsUri { get; set; }
        public IEnumerable<CompanyName> Names { get; set; } = null!;
        public IEnumerable<Liquidation>? Liquidations { get; set; }
        [NotMapped]
        public IEnumerable<CompanyName>? AuxiliaryNames
        {
            get => Names.AsQueryable().Where(n => n.Order > 0);
        }
        public IEnumerable<Address>? Addresses { get; set; }
        public IEnumerable<CompanyForm>? CompanyForms { get; set; }
        public IEnumerable<BusinessLine>? BusinessLines { get; set; }
        public IEnumerable<Language>? Languages { get; set; }
        public IEnumerable<RegistedOffice>? RegistedOffices { get; set; }
        public IEnumerable<ContactDetail>? ContactDetails { get; set; }
        public IEnumerable<RegisteredEntry>? RegisteredEntries { get; set; }
        public IEnumerable<BusinessIdChange>? BusinessIdChanges { get; set; }
    }

    public enum Source
    {
        Common,
        Prh,
        Vero,
        Ytj
    }

    public enum CompanyFormType
    {
        OY,
        OYJ,
        OK,
        AOY,
        VOJ
    }
}

File: client/src/types/company.ts
import { Address } from "cluster";
import { CompanyName } from "./companyName";
import { Liquidation } from "./liquidation";
import { CompanyForm } from "./companyForm";
import { BusinessLine } from "./businessLine";
import { Language } from "./language";
import { RegistedOffice } from "./registedOffice";
import { ContactDetail } from "./contactDetail";
import { RegisteredEntry } from "./registeredEntry";
import { BusinessIdChange } from "./businessIdChange";

export interface Company {
    businessId: string;
    name: string;
    registrationDate: string;
    companyForm: CompanyFormType | null;
    detailsUri: string | null;
    names: CompanyName[];
    liquidations: Liquidation[] | null;
    auxiliaryNames: CompanyName[] | null;
    addresses: Address[] | null;
    companyForms: CompanyForm[] | null;
    businessLines: BusinessLine[] | null;
    languages: Language[] | null;
    registedOffices: RegistedOffice[] | null;
    contactDetails: ContactDetail[] | null;
    registeredEntries: RegisteredEntry[] | null;
    businessIdChanges: BusinessIdChange[] | null;
}

export interface CompanyResponse {
    results: Company[]
}

export enum Source {
    Common,
    Prh,
    Vero,
    Ytj
}

export enum CompanyFormType {
    OY,
    OYJ,
    OK,
    AOY,
    VOJ
}